src/main/scala/expressd/ExpressD.scala

```diff
@@ -893,9 +902,17 @@ object ExpressD {
             val misc2R = right._4(i)
             val misc3R = right._5(i)
 
-            val pairLength = rightStart - leftStart + 1
+            //val pairLength = rightStart - leftStart + 1
+            val pairLength = Math.abs(rightStart - leftStart) + 1
```

Comment:
I assumed this is what was intended.  Sometimes right fragment is listed first in SAM input, perhaps because it is not a sorted BAM.